packages/components/inputs/rich-text-utils/src/editor.types.ts

```diff
@@ -0,0 +1,114 @@
+import { ReactNode } from 'react';
+
+type TSize = {
+  size: number;
+};
+
+type TData = {
+  get: (key: string) => TSize;
+};
+
+type TValue = {
+  blocks: TBlocks;
+  document: TDocument;
+  selection: {
+    isFocused: boolean;
+  };
+  activeMarks:
+    | {
+        some: (ascertMarkType: (mark: TMark) => boolean) => void;
+      }
+    | TMark[];
+  data: TData;
+};
+
+type TWrapBlock = {
+  wrapBlock: (blockTag: string) => void;
+};
+type TBlocks = {
+  size?: number;
+  some: (block: unknown) => boolean | void;
+  first: () => {
+    key: string;
+    type: string;
+  };
+  map: (node: ReactNode) => {
+    toArray: () => string[];
+  };
+};
+
+type TNodes = {
+  map: (node: ReactNode) => {
+    toArray: () => string[];
+  };
+};
+
+type TDocument = {
+  getClosest: (
+    block: { key: unknown },
+    closest: (parent: { type: string }) => boolean
+  ) => boolean;
+  getParent?: (parentParam: string) => {
+    type: string;
+  };
+  text?: string;
+  nodes?: TNodes;
+};
+
+type TNodeTextsResult = [{ key: string; text: string }, string];
+
+export type TMark = {
+  type: string;
+};
+
+export type TNode = {
+  texts: (nodeParam?: Record<string, string>) => TNodeTextsResult[];
+};
+
+export type TOptions = {
+  hotkey: string;
+};
+
+export type TEditor = {
```

Comment:
Do we need to define the types for this manually? Isn't there a type exposed from the library? 🤔

Reply:
If I may: it seems that `Editor` class from https://www.npmjs.com/package/@types/slate-react/v/0.22.9 is the closest to what we expect from `TEditor` type, but if I'm not mistaken this class would have to be extended manually anyway 🤔

Reply:
So Slate has [TS support](https://docs.slatejs.org/concepts/12-typescript) but not in the current version we are using.

I opened a separate issue for that: #2120 

So let's keep the "custom types" for now.